crates/holochain/src/core/workflow/app_validation_workflow.rs

```diff
@@ -1,4 +1,98 @@
-//! The workflow and queue consumer for sys validation
+//! Holochain workflow to validate all incoming DHT operations with an
+//! app-defined validation function.
+//!
+//! Triggered by system validation, this workflow iterates over a list of
+//! [`DhtOp`]s that have passed system validation, validates each op, updates its validation status
+//! in the database accordingly, and triggers op integration if necessary.
+//!
+//! ### Sequential validation
+//!
+
+// Even though ops are validated in sequence, they could all be validated in parallel too with the same result.
+// All actions are written to the database straight away in the incoming dht ops workflow and do not require validation to be available for validating other ops. See https://github.com/holochain/holochain/issues/3724
+
+//! Ops are validated in sequence based on their op type and the timestamp they
+//! were authored (see [`OpOrder`] and [`OpNumericalOrder`]). Validating one op
+//! after the other with this ordering was chosen so that ops that depend on earlier
+//! ops will be validated after the earlier ops, and therefore have a higher chance
+//! of being validated successfully. An example is an incoming delete
+//! op that depends on a create op. Validated in order of their authoring, the
+//! create op is validated first, followed at some stage by the delete op. If
+//! the validation function references the original action when validating
+//! delete ops, the create op will have been validated and is available in the
+//! database. Otherwise the delete op could not be validated and its dependency,
+//! the create op, would be awaited first.
+//!
+//! ### Op validation
+//!
+//! For each op the [corresponding app validation function](https://docs.rs/hdi/latest/hdi/#data-validation)
+//! is executed. Entry and link CRUD actions, which the ops are derived from, have been
+//! written with a particular integrity zome's entry and link types. Thus for
+//! op validation, the validation function of the same integrity zome must be
+//! used. Ops that do not relate to a specific entry or link like [`DhtOp::RegisterAgentActivity`]
+//! or non-app entries like [`EntryType::CapGrant`] are validated with all
+//! validation functions of the DNA's integrity zomes.
+//!
+//! Having established the relevant integrity zomes for validating an op, each
+//! zome's validation callback is invoked.
+//!
+//! ### Outcome
+//!
+//! An op can be valid or invalid, which is considered "validated", or it could not be
```

Comment:
> An op can be valid or invalid, which is considered "validated"

This has always been kind of confusing. The meaning of "validate" is "to establish the soundness, accuracy, or legitimacy of", so it seems to apply more to positive validation.

It would be nice if we could call something "integrated" if it's gone through the validation workflow, "validated" if it passed and "rejected" if it failed. But that only works if we remove the integration workflow (which we might decide to do).

Reply:
It's certainly used in a sense of declaring something valid. The literal meaning is fine, I validate something and can turn out to be valid or invalid.

Something more general like "checked" comes to mind. Although if I say "an op can be valid or valid, which considered 'checked'", it has the same feel to it.

Reply:
I recognize that this question doesn't have to affect this PR, since this PR is just documenting what already is, including our language. Maybe we can follow up elsewhere if anyone else finds this confusing.